src/Lecture2.hs

```diff
@@ -52,7 +52,10 @@ zero, you can stop calculating product and return 0 immediately.
 84
 -}
 lazyProduct :: [Int] -> Int
-lazyProduct = error "TODO"
+lazyProduct list = case list of
+    (0 : _) -> 0
+    [num] -> num
+    _ -> head list * lazyProduct (tail list)
```

Comment:
Usually, when you call `head` and `tail` on the same list, it's a good chance you want to pattern match on a list instead:

```suggestion
    x : xs -> x * lazyProduct xs
```

Reply:
Right, true! I thought `x : xs` wasn't very expressive but I suppose it's clear and understandable to people who write Haskell, I'll use it.